frontend/src/Monitor/Leaderboard/Charts/ResourceProduction.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import Chart from 'Monitor/Charts/Chart';
+import LineChart from 'Monitor/Charts/LineChart';
+import {selectNested} from 'Monitor/Charts/Util';
+import {useLeaderboardContext} from '../Context';
+import {getTimeseries, avgReducer, sumReducer} from '../Selectors';
+import {registerChart} from 'Monitor/Charts/ChartRegistry';
+import {LeaderboardChartProps} from './Types';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import {makeStyles} from '@material-ui/core/styles';
+import { Checkbox, ListItemText } from '@material-ui/core';
+
+// Like `Object.keys` but typed. TODO: move to reusable commons?
+type KeyValue<T extends Record<string, unknown>, K extends keyof T = keyof T> = [K, T[K]]
+export const recordEntries = <T extends Record<string, unknown>>(o: T) => Object.entries(o) as KeyValue<T>[];
+export const recordKeys = <T extends Record<string, unknown>>(o: T) => Object.keys(o) as (keyof T)[];
+export const recordValues = <T extends Record<string, unknown>>(o: T) => Object.values(o) as T[keyof T][];
+
+const Name = 'Resource Production';
+
+const useStyles = makeStyles((theme) => ({
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+}));
+
+enum RESOURCE_PART {
+    // Energy = "Energy",  // TODO: data from BE
```

Comment:
If you would like to add this data to the BE in this PR it's only a few lines of change to: https://github.com/benreid24/Stellaru/blob/master/backend/engine/extraction/extractors/economy_extractor.py#L153

The energy amounts are already available there, you'd just have to add them to the GDP dict that is returned. The entire GDP dict is added to the empire summaries so it will get picked up.